azext_edge/edge/providers/check/base/deployment.py

```diff
@@ -24,22 +25,54 @@
 # TODO: unit test
 
 
-def check_pre_deployment(
-    result: Dict[str, Any],
-    as_list: bool = False,
-) -> None:
-    result["preDeployment"] = []
+def validate_cluster_prechecks(acs_config: Optional[dict] = None) -> None:
+    pre_checks = check_pre_deployment(acs_config=acs_config)
+    errors = {}
+    for check in pre_checks:
+        for target in check["targets"]:
+            for namespace in check["targets"][target]:  # for all prechecks, namespace is currently _all_
+                for idx, check_eval in enumerate(check["targets"][target][namespace]["evaluations"]):
+                    if check_eval["status"] not in NON_ERROR_STATUSES:
+                        # TODO - relies on same order and count of conditions / evaluations
+                        expected_condition = (
+                            check["targets"][target][namespace]["conditions"][idx]
+                            if idx < len(check["targets"][target][namespace]["conditions"])
+                            else None
+                        )
+                        if not errors.get(target):
```

Comment:
errors could be `errors = defaultdict(list)` and then this condition is not necessary.